src/mds/Server.h

```diff
@@ -327,6 +327,14 @@ class Server {
   bool terminating_sessions = false;
 
   std::set<client_t> client_reclaim_gather;
+  // record laggy clients due to laggy OSDs
+  std::set<client_t> laggy_clients;
```

Comment:
client_t is better as we're talking about client eviction, using client_t gives the ID of the client that is laggy i also need to store non responding clients in `Server::evict_cap_revoke_non_responders` where the non responders are fetched using `mds->locker->get_late_revoking_clients(cap_revoke_eviction_timeout)` whose type of client_t

Reply:
Fair enough, however, `to_evict` seems redundant.

Reply:
`to_evict` is needed. If in case `defer_client_eviction` is false(a bool value that depends on the config option `defer_client_eviction_on_laggy_osds` && if any OSD is laggy), `to_evict` is used to evict the clients. If you're pointing to unifying both `laggy_clients` and `to_evict` then it might not be a good approach.

Although, I agree to make this a private member.